backend/controllers/searchDownload.js

```diff
@@ -0,0 +1,35 @@
+const logger = require('../log')
+
+const downloadFactory = (searchApi, json2CsvParse) => {
+  const downloadResults = async (req, res) => {
+    const { user, groupCode, roleCode, status } = req.query
+
+    const searchResults = await searchApi({
+      locals: res.locals,
+      user,
+      groupCode,
+      roleCode,
+      status,
+      pageNumber: 0,
+      pageSize: 10000,
+      pageOffset: 0,
+    })
+
+    try {
+      const csv = json2CsvParse(searchResults)
+      res.header('Content-Type', 'text/csv')
+      res.attachment('user-search.csv')
+      return res.send(csv)
+    } catch (err) {
+      logger.error(err)
+      res.writeHead(400, { 'Content-Type': 'text/plain' })
```

Comment:
why the `400` here?

Reply:
It was just the example I copied from and it worked ok.  Didn't actually try any other codes - are you thinking it should be 5xx?

Reply:
certainly, since it is a server error rather than a bad request